[docs] Describe `--cache` (#5969)
DevExpress/testcafe: docs/articles/documentation/reference/command-line-interface.md

testcafe chrome my-tests --compiler-options typescript.options.lib=lib.es5.d.ts,lib.webworker.d.ts;typescript.typesRoot='this value contains spaces'
```

> The `--compiler-options` parameter accepts the compiler options listed in the [official TypeScript documentation](https://www.typescriptlang.org/docs/handbook/compiler-options.html), and two additional options: `typescript.configPath` and `typescript.customCompilerModulePath`.

Reviewer: ```suggestion
The `--compiler-options` parameter accepts the compiler options listed in the [official TypeScript documentation](https://www.typescriptlang.org/docs/handbook/compiler-options.html) and two additional options: `typescript.configPath` and `typescript.customCompilerModulePath`.
```

*Related configuration file property*: [compilerOptions](configuration-file.md#compileroptions).
*Related API method*: [runner.compilerOptions](testcafe-api/runner/compileroptions.md)

### --cache

> If enabled, TestCafe proxy caches webpage dependencies (like styles and scripts) for webpages visited during tests. When TestCafe revisits the page, it pulls assets from the cache instead of requesting them from the server.

Reviewer 2: Uniform phrasing

```suggestion
If enabled, the TestCafe proxy caches webpage assets (such as stylesheets and scripts) for the webpages that it processes. The next time the proxy accesses the page, it pulls assets from its cache instead of requesting them from the server.
```

```sh
testcafe chrome my-tests --cache
```

> TestCafe emulates the current browser's caching mechanism. TestCafe only caches resources that the browser itself would cache if run without TestCafe.

Reviewer 3: `native browser's caching mechanism`

Author: > TestCafe emulates the current browser's native caching mechanism.

> TestCafe emulates the native caching mechanism of the browser. For example, in *Chrome* TestCafe only caches resources that *Chrome* itself would cache if run without TestCafe.

Reviewer 2: pls don't abuse the word 'mechanism'

```suggestion
TestCafe emulates browsers' native caching behavior. For example, when working with *Chrome*, TestCafe only caches the resources that *Chrome* itself would cache in a real-life scenario.
```

Author: Testing with TestCafe is as `real-life` as running the browser without TestCafe.

`When working with` is an unnecessary complication. `In` should suffice

```suggestion
TestCafe emulates the browser's native caching behavior. For example, in *Chrome* TestCafe only caches resources that *Chrome* itself would cache if run without TestCafe.
```

> TestCafe caches scripts, styles, fonts, and other web assets, except for HTML. TestCafe doesn't cache HTML because that could interfere with [user roles](../guides/advanced-guides/authentication.md#user-roles). Only assets up to **5 MB** in size are stored in cache.

Reviewer 2: ```suggestion
TestCafe caches scripts, stylesheets, fonts, and other page assets up to **5 MB** in size. TestCafe doesn't cache HTML because that could lead to issues with [user role recall](../guides/advanced-guides/authentication.md#user-roles).
```

Author: `Web assets` is a [common term](https://www.google.com/search?newwindow=1&ei=ncQwYK6-Dsqk3APrlpXgAg&q=web+asset&oq=web+asset&gs_lcp=Cgdnd3Mtd2l6EAMyBQgAEJECMgIIADIECAAQQzICCAAyAggAMgIIADIGCAAQFhAeMgYIABAWEB4yBggAEBYQHjIICAAQFhAKEB46BwgAEEcQsAM6CAguEMcBEKMCOgoILhDHARCjAhBDOgQIABAKOgQIABANOgYIABANEB5Q1QxYmhxgpR1oAnACeACAAZABiAGXCJIBAzguM5gBAKABAaoBB2d3cy13aXrIAQjAAQE&sclient=gws-wiz&ved=0ahUKEwjugOGggvjuAhVKEncKHWtLBSwQ4dUDCA0&uact=5). I've barely seen `Page assets` used anywhere.

`up to **5 MB** in size` sounds like we have 5 MB of cache per page. Actually, 5 MB is the size limit for an individual asset. Should ask correctors about this one.

`that could lead to issues with user role **recall**` - I'm not sure it has to do with recall specifically. 
@miherlosev can you confirm?

```suggestion
TestCafe caches scripts, stylesheets, fonts, and other web assets up to **5 MB** in size. TestCafe doesn't cache HTML because that could break [user roles](../guides/advanced-guides/authentication.md#user-roles).
```

> To decrease test run time, enable server-side caching when you test asset-heavy web applications.

Reviewer: I'm not sure `asset-heavy` is right term because it seems quite informal

Author: Admittedly, this is an informal and imprecise term. The problem here is that we don't know how `heavy` the page needs to be to cause problems (it depends on many factors outside of our control). 

So it's on the user to know that their app is `asset-heavy`. 

We can't give a precise number here (or even an approximate one). An alternative is:

```suggestion
If the tested application loads a lot of assets, enable server-side caching to decrease test run time.
```

But, again, `a lot` is not precise nor formal. I think we can allow some informality here.

Author: Does this work?

```suggestion
If the tested application loads many heavy assets, enable server-side caching to decrease test run time.
```

> To decrease test run time, enable `--cache` when you test web applications that require a lot of assets.

Reviewer 2: Assets != requirements
```suggestion
To decrease test run time, enable `--cache` when you test asset-rich web applications.
```

Author: We've [been there](https://github.com/DevExpress/testcafe/pull/5969#discussion_r578197157)

Plus, assets themselves are not a problem, requesting assets is.

Author: ```suggestion
To decrease test run time, enable `--cache` when you test web applications that load a lot of assets.
```

Reviewer 2: "A lot" is very subjective, and refers solely to the number of assets requested by the page. The word 'heavy', on the other hand, implies that the system is strained under an increased load. The word 'rich' implies a sort of saturation. Both are a much, much better fit for this sentence than the phrase 'apps that require a lot of assets'. 

>Plus, assets themselves are not a problem, requesting assets is.

If this is the case, you should've specified that you're talking about *remote* assets in particular. 
Inline SVG images are still assets.

```suggestion
To decrease test run time, enable `--cache` when you test web applications that rely on a large amount of remote assets.
```

Author: As you pointed out, the number of assets is not a problem: their weight is. 

`The word 'heavy', on the other hand, implies that the system is strained under an increased load.` - that's exactly what we're going for in my opinion.

I still think this phrasing is best:

```suggestion
To decrease test run time, enable server-side caching when you test asset-heavy web applications.
```

> > Important! Support for server-side caching is experimental. Disable the `--cache` flag if you run into compatibility issues with your tests.

Reviewer: ```suggestion
> Important! Support for server-side caching is experimental. Disable the `--cache` flag if you have compatibility issues with your tests.
```

> > Important! The `--cache` feature is experimental. Turn it off if you experience instabilities in your tests.

Reviewer 2: `--cache` is not a feature, it's a flag.  +Phrasing consistency

```suggestion
> Important! Support for server-side caching is experimental. Disable the `--cache` flag if you run into compatibility issues with your tests.
```
